File: Data Structures & Algorithms/products-of-array-discluding-self/submission-1.java
class Solution {
    public int[] productExceptSelf(int[] nums) {
        int[] temp = new int[nums.length];
        int perfix = 1;
        for(int i = 0 ; i <nums.length ; i++){
            temp[i] = perfix;
            perfix *= nums[i];
        }
        int[] res = new int[nums.length];
        int suf =1;
        for(int i = nums.length-1 ; i >= 0 ; i--){
            res[i] = temp[i] * suf;
            suf *= nums[i];
        }
        return res;
    }
}  

File: Data Structures & Algorithms/three-integer-sum/submission-1.cs
public class Solution {
    public List<List<int>> ThreeSum(int[] nums) {
        Array.Sort(nums);
        if(nums.Length == 0 || nums[0] > 0)
            return new List<List<int>>();

        var res = new List<List<int>>() ;

        for(int i=0 ; i < nums.Length; i++){
            if (i > 0 && nums[i] == nums[i - 1]) continue;

            int lp = i+1;
            int up = nums.Length-1;
            while(lp < up){
                int temp = nums[lp] + nums[up] + nums[i];
                if( temp == 0){
                    res.Add(new int[]{nums[lp] , nums[up] , nums[i]}.ToList());
                    lp++;
                    while (lp < up && nums[lp] == nums[lp - 1]) lp++;
                }
                else if( temp > 0)
                    up--;
                else
                    lp++;
            }
        }

        return res;

    }
}
